specification/resources/resource-manager/Microsoft.Authorization/preview/2023-03-01-preview/privateLinks.json

```diff
@@ -0,0 +1,832 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ResourcePrivateLinkClient",
+    "version": "2023-03-01-preview",
+    "description": "Provides operations for managing private link resources"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Management/managementGroups/{groupId}/providers/Microsoft.Authorization/privateLinkAssociations/{plaId}": {
+      "put": {
+        "tags": [
+          "PrivateLinkAssociation"
+        ],
+        "operationId": "PrivateLinkAssociation_Put",
+        "description": "Create a PrivateLinkAssociation",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/GroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkAssociationIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateLinkAssociation"
+            },
+            "description": "Parameters supplied to create the private link association."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok - Returns information about the new private link association update.",
+            "schema": {
+              "$ref": "#/definitions/PrivateLinkAssociation"
+            }
+          },
+          "201": {
+            "description": "Created - Returns information about the new private link association.",
+            "schema": {
+              "$ref": "#/definitions/PrivateLinkAssociation"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
```

Comment:
this looks slightly off - shouldn't it be a reference directly to 
ErrorResponse in common types, instead of defining a CloudError type here which contains an ErrorResponse inside of it??

Reply:
Not adding it here, Its present on version 2020-05-01 and if we try to change it to use common type V5 Error. The PR marked as breaking change.

Reply:
Making this change is breaking. Thinking if we should revert it to the previous one. Error we get is. 

The new version is missing a property found in the old version. Was 'error' renamed or removed?
New: [common-types/resource-management/v1/types.json#L289:7](https://github.com/Azure/azure-rest-api-specs/blob/8074086a782d606cb690eeb25c2dcbf823efd92e/specification/common-types/resource-management/v1/types.json#L289:7)
Old: [Microsoft.Authorization/stable/2020-05-01/privateLinks.json#L598:7](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Authorization/stable/2020-05-01/privateLinks.json#L598:7)